src/ConformanceTests.jl

```diff
@@ -0,0 +1,71 @@
+module ConformanceTests
+
+using ..AbstractAlgebra
+
+# helper
+function equality(a, b)
+  if is_exact_type(typeof(a)) && is_exact_type(typeof(b))
+     return a == b
+  else
+     return isapprox(a, b)
+  end
+end
+
+function equality_up_to_units(a, b)
```

Comment:
Didn't realize we had this function; so this remarks is not directly for this PR, but I also don't forget: I think we can delete it and replace all calls to it by calls to `is_associated`